plaidml/bridge/openvino/ops/deformable_convolution.cpp

```diff
@@ -0,0 +1,389 @@
+  offset_w = op::transpose(offset_w, make_tuple<int64_t>({0, 1, 5, 2, 6, 3, 7, 4}));
+  offset_d = op::reshape(offset_d, make_tuple<int64_t>({N, DG, 1, OFF_D * F_D, OFF_H * F_H, OFF_W * F_W}));
+  offset_h = op::reshape(offset_h, make_tuple<int64_t>({N, DG, 1, OFF_D * F_D, OFF_H * F_H, OFF_W * F_W}));
+  offset_w = op::reshape(offset_w, make_tuple<int64_t>({N, DG, 1, OFF_D * F_D, OFF_H * F_H, OFF_W * F_W}));
+  offset_d = op::broadcast(offset_d, {N, DG, CI / DG, NEW_D, NEW_H, NEW_W}, {0, 1, 2, 3, 4, 5});
+  offset_h = op::broadcast(offset_h, {N, DG, CI / DG, NEW_D, NEW_H, NEW_W}, {0, 1, 2, 3, 4, 5});
+  offset_w = op::broadcast(offset_w, {N, DG, CI / DG, NEW_D, NEW_H, NEW_W}, {0, 1, 2, 3, 4, 5});
+  offset_d = op::reshape(offset_d, make_tuple<int64_t>({N, CI, NEW_D, NEW_H, NEW_W}));
+  offset_h = op::reshape(offset_h, make_tuple<int64_t>({N, CI, NEW_D, NEW_H, NEW_W}));
+  offset_w = op::reshape(offset_w, make_tuple<int64_t>({N, CI, NEW_D, NEW_H, NEW_W}));
+  // Define depth, hight and width of input.
+  edsl::Tensor index_d_0 = edsl::index({edsl::TensorDim(F_D), edsl::TensorDim(OFF_D)}, static_cast<size_t>(0));
+  edsl::Tensor index_d_1 = edsl::index({edsl::TensorDim(F_D), edsl::TensorDim(OFF_D)}, static_cast<size_t>(1));
+  edsl::Tensor index_d = index_d_0 * dilations[0] + index_d_1 * strides[0] - pad_befores[0];
+  index_d = op::transpose(index_d, make_tuple<int64_t>({1, 0}));
+  index_d = op::reshape(index_d, make_tuple<int64_t>({1, 1, OFF_D * F_D, 1, 1}));
+  index_d = op::broadcast(index_d, {1, 1, NEW_D, NEW_H, NEW_W}, {0, 1, 2, 3, 4});
+  edsl::Tensor index_h_0 = edsl::index({edsl::TensorDim(F_H), edsl::TensorDim(OFF_H)}, static_cast<size_t>(0));
+  edsl::Tensor index_h_1 = edsl::index({edsl::TensorDim(F_H), edsl::TensorDim(OFF_H)}, static_cast<size_t>(1));
+  edsl::Tensor index_h = index_h_0 * dilations[1] + index_h_1 * strides[1] - pad_befores[1];
+  index_h = op::transpose(index_h, make_tuple<int64_t>({1, 0}));
+  index_h = op::reshape(index_h, make_tuple<int64_t>({1, 1, 1, OFF_H * F_H, 1}));
+  index_h = op::broadcast(index_h, {1, 1, NEW_D, NEW_H, NEW_W}, {0, 1, 2, 3, 4});
+  edsl::Tensor index_w_0 = edsl::index({edsl::TensorDim(F_W), edsl::TensorDim(OFF_W)}, static_cast<size_t>(0));
+  edsl::Tensor index_w_1 = edsl::index({edsl::TensorDim(F_W), edsl::TensorDim(OFF_W)}, static_cast<size_t>(1));
+  edsl::Tensor index_w = index_w_0 * dilations[2] + index_w_1 * strides[2] - pad_befores[2];
+  index_w = op::transpose(index_w, make_tuple<int64_t>({1, 0}));
+  index_w = op::reshape(index_w, make_tuple<int64_t>({1, 1, 1, 1, OFF_W * F_W}));
+  index_w = op::broadcast(index_w, {1, 1, NEW_D, NEW_H, NEW_W}, {0, 1, 2, 3, 4});
+  // Get deformabled index.
+  edsl::Tensor new_index_d = offset_d + index_d;
+  edsl::Tensor new_index_h = offset_h + index_h;
+  edsl::Tensor new_index_w = offset_w + index_w;
+  // Get deformable input tensor.
+  edsl::Tensor deform_input = edsl::gather(I, new_index_d).axis(-3);
```

Comment:
Please check if we can use the latest gather_nd https://github.com/plaidml/plaidml/pull/1695/files here. @haoyouab @DingZhangIntel

Reply:
Ok, I will try.